src/combined-selectors/reporting-config-search-results.ts

```diff
@@ -41,16 +67,37 @@ function searchReportingConfig(
 			feature.keywords?.some( ( keyword ) => includesIgnoringCase( keyword, searchTerm ) );
 
 		if ( includesIgnoringCase( feature.name, searchTerm ) || keywordsIncludeSearchTerm() ) {
-			searchResults.features.add( featureId );
-
-			const feature = features[ featureId ];
-			if ( feature.parentType === 'product' ) {
-				searchResults.products.add( feature.parentId );
-			} else {
-				searchResults.featureGroups.add( feature.parentId );
-				const parentFeatureGroup = featureGroups[ feature.parentId ];
-				searchResults.products.add( parentFeatureGroup.productId );
+			addFeatureAndParents( featureId );
+		}
+	}
+
+	// Finally, search for the search term in the description of the products and feature areas
+	const searchTermTokens = tokenizeAndNormalize( searchTerm );
+	const scores: Record< string, { type: TaskParentEntityType; score: number } > = {};
+
+	for ( const token of searchTermTokens ) {
+		const matchingEntities = invertedIndex[ token ] || [];
+
+		for ( const { type, id, weight } of matchingEntities ) {
+			if ( ! scores[ id ] ) {
+				scores[ id ] = { type, score: 0 };
 			}
+			scores[ id ].score += weight;
```

Comment:
**Suggestion:** I might be misunderstanding, but while the block below is now id-safe, I think this part may still not be id-safe? That is, if two entities, across two different types (like a feature and a featureGroup) both had the same ID, they would both get counted in the same score, right? I think we need to make sure the dictionary entries are all unique per entity!

I think there's two ways you go do this:

1) You could make the scores dictionary effectively three dictionaries by type, then just keep using ID:

```typescript
const scores = {
    products: {
        [id]: { } // stuff
    }
    featureGroups: {} // etc
    features: {} // etc
}
```

2) Or you could make the key unique by combining the `type` and `id`! But then you'd also probably want to store the raw ID:

`scores[ type + id ] = { type, id, score: 0 }`

Reply:
Got it! I've changed the implementation here: [3c09993](https://github.com/Automattic/bugomattic/pull/80/commits/3c099936a0db1faa14d76daf32baa8a980c1b22b)